### PR TITLE
Update master branch to Pastas 1.5

### PR DESCRIPTION
**Pull request by raoulcollenteur**
# Short Description

Big update for Pastas to version 1.5. 

# Checklist before PR can be merged:
- [x] is documented
- [x] Format code with [Black formatting](https://black.readthedocs.io)
- [x] type hints for functions and methods
- [x] tests added / passed
- [x] Example Notebook (for new features)
- [x] Remove output for all notebooks with changes

### CODE REVIEW

**Review by mbakker7 — APPROVED** [review]
Ready to go. Some minor things in Codacy that can be fixed next time.

File: doc/examples/interactive_plots.ipynb
Comment (dbrakenhoff) [inline doc/examples/interactive_plots.ipynb:1]: We should add the stderr kwarg to the plotly and bokeh results plots as well probably.

File: doc/examples/prepare_timeseries.ipynb
Comment (dbrakenhoff) [inline doc/examples/prepare_timeseries.ipynb:1]: Should be uncommented:
```python
# s_pd2.plot(ax=ax, marker="x", ms=10, label="pandas_equidistant_nearest")
# s_pd3.plot(ax=ax, marker="^", ms=8, label="pandas_equidistant_asfreq")
# s_pd1.plot(ax=ax, marker="+", ms=16, label="pandas_equidistant_sample")
```

File: pastas/__init__.py [hunk 1]
```diff
@@ -43,6 +45,19 @@
...
 
+
+def custom_formatwarning(msg, *args, **kwargs):
+    # ignore everything except the message
+    return str(msg)
+
+
+warnings.formatwarning = custom_formatwarning
+warnings.warn(
+    """DeprecationWarning: As of Pastas 1.5, no noisemodel is added to the pastas Model class by default anymore. To solve your model using a noisemodel, you have to explicitely add a noisemodel to your model before solving. For more information, and how to adapt your code, please see this issue on GitHub: https://github.com/pastas/pastas/issues/735""",
```
Comment (dbrakenhoff) [inline pastas/__init__.py:56]: typo: explicitly

File: pastas/model.py [hunk 2]
```diff
@@ -140,8 +142,27 @@ def __init__(
...
             self.add_constant(constant)
-        if noisemodel:
-            self.add_noisemodel(NoiseModel())
+
+        if noisemodel is not None:
+            if noisemodel is True:
+                msg = (
+                    "The noisemodel argument is deprecated and will be removed in Pastas "
+                    "version 2.0.0. The new default is that no noisemodel is added "
+                    "anymore and a noisemodel has to be added explicitely to a Pastas "
```
Comment (dbrakenhoff) [inline pastas/model.py:151]: typo: explicitly

**Review by dbrakenhoff — APPROVED** [review]
Some very minor comments, one comment about stderrs in interactive plots that can easily be picked up in the next release.

File: doc/examples/interactive_plots.ipynb
Comment (raoulcollenteur) [reply doc/examples/interactive_plots.ipynb:1]: I created a new issue for 1.6

**Comment by raoulcollenteur** [comment]
Thanks for reviews! Ready to merge after merging #743 into Dev, to add the banner on the homepage as suggested by @mbakker7.

**Comment by mbakker7** [comment]
In case you are still tinkering with the README: All the little icons (CI, pypi, license, etc.) now appear on a separate line. That takes up a lot of vertical space. I think it looks better if many are put on one line as it used to be. Not important, but it is our opening page, so we want it to look nice.

**Pull request merged**